### PR TITLE
Testing moves tracker

### PR DESCRIPTION
**Pull request by Lewis-Kelley**

### CODE REVIEW

File: 2D Simulation/movesTracker.js [hunk 1]
```diff
@@ -1,99 +1,114 @@
...
+    return "Block id: " + action.id + " "
+        + "Letter: " + action.letter + " "
+        + "Color: " + action.color;
+}
+
+function _export_position(left_pos, top_pos) {
+    return "(" + left_pos + "," + top_pos + ")";
+}
+
+if (typeof module !== 'undefined'
```
Comment (jcolemang) [inline 2D Simulation/movesTracker.js:103]: As long as we're doing stuff like this would it make sense to use actual js modules? We would probably need to add in babel or something

File: 2D Simulation/spec/movesTrackerSpec.js [hunk 2]
```diff
@@ -0,0 +1,395 @@
+describe("movesTracker", () => {
+    let imported = require('../movesTracker.js');
```
Comment (jcolemang) [inline 2D Simulation/spec/movesTrackerSpec.js:2]: I thought `require` was a RequireJS thing? I haven't done much with JavaScript modules.

**Review by jcolemang — APPROVED** [review]

File: 2D Simulation/spec/movesTrackerSpec.js (hunk 2 above)
Comment (Lewis-Kelley) [reply 2D Simulation/spec/movesTrackerSpec.js:2]: Surprisingly no! This is actually Node's `require`. RequireJS has much different formatting (it's more akin to dependency injection than an import statement).

File: 2D Simulation/movesTracker.js (hunk 1 above)
Comment (Lewis-Kelley) [reply 2D Simulation/movesTracker.js:103]: SIgh... You have no idea how much I'd like to do that.

First of all, setting everything up to use modules would take a good deal of refactoring effort. I think it would be worth it, but I want to push through the E2E tests before we do anything that exciting.

Secondly, for whatever reason, Jasmine does not like `import` / `export` keywords (it says unrecognized token or something like that), so yes, we'd have to use babel for that, which I haven't tested much with yet and as such still might not cooperate.

The last option might be RequireJS, which Jasmine also does not seem to appreciate very much when run through the command line. Instead, you have to use their `SpecRunner.html` and load RequireJS into it in order for it to work...
 
Also of note is that the HTML would no longer have access to global functions (like the various buttons wouldn't see their callbacks anymore), and as such the JavaScript would have to use jQuery to attach the functions to the buttons manually. I still think this is worthwhile, mind, it's just that this will take a concerted effort.

TL;DR: Hopefully soon?

File: 2D Simulation/spec/movesTrackerSpec.js (hunk 2 above)
Comment (kgroble) [inline 2D Simulation/spec/movesTrackerSpec.js:2]: This isn't very important either way, but is there any reason these all aren't `const`s instead of `let`s?

**Review by kgroble — APPROVED** [review]

File: 2D Simulation/spec/movesTrackerSpec.js (hunk 2 above)
Comment (Lewis-Kelley) [reply 2D Simulation/spec/movesTrackerSpec.js:2]: No... I honestly forgot that was a part of JavaScript... I'll change them as they come up.
